org.eclipse.gef/src/org/eclipse/gef/EditPartViewer.java

```diff
@@ -281,7 +281,22 @@ interface Conditional {
 	 *
 	 * @return the registry map
 	 */
-	Map getEditPartRegistry();
+	Map<? extends Object, ? extends EditPart> getEditPartRegistry();
+
+	/**
+	 * Convenience method to look up an edit part for a given model element in the
+	 * EditPart registry.
+	 *
+	 * See also {@link #getEditPartRegistry()} for details on the EditPart registry.
+	 *
+	 * @param model the model object for which an EditPart is looked up
+	 * @return the edit part or null if for the given model no EditPart is
+	 *         registered
+	 * @since 3.19
+	 */
+	default EditPart getEditPartForModel(Object model) {
+		return getEditPartRegistry().get(model);
```

Comment:
I get your point. However Java does not allow that default methods of interfaces are final. I can only add a @nooverride. Would this be sufficient.

Reply:
Then I would only keep the signature in the interface and move the actual implementation to the AbstractEditPartViewer (where it can then be final). The documentation explicitly says that clients should never implement the interface directly.

Reply:
done